From cbc17c0ef9647dbfa9aaa56da99eaf8f8a6d6469 Mon Sep 17 00:00:00 2001
From: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Date: Thu, 30 Oct 2014 10:54:08 +0530
Subject: [PATCH 724/741] usb: pm: Fix pm defect in rcpm driver for usb ip

commit e728911a10b67bb53ea68d66fb3103951e6fd298 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Pass device-tree usb node offset to each usb device structure.
This is required by rcpm driver to get mask setting for
ip exception register to make/remove usb as wake-up source
for sleep/deep-sleep

Signed-off-by: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/core/usb.c           | 1 +
 drivers/usb/host/fsl-mph-dr-of.c | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/drivers/usb/core/usb.c b/drivers/usb/core/usb.c
index 7fcb9f782931..91922175dee8 100644
--- a/drivers/usb/core/usb.c
+++ b/drivers/usb/core/usb.c
@@ -613,6 +613,7 @@ struct usb_device *usb_alloc_dev(struct usb_device *parent,
 	dev->dev.dma_mask = bus->sysdev->dma_mask;
 	dev->dev.dma_pfn_offset = bus->sysdev->dma_pfn_offset;
 	set_dev_node(&dev->dev, dev_to_node(bus->sysdev));
+	dev->dev.of_node = bus->controller->of_node;
 	dev->state = USB_STATE_ATTACHED;
 	dev->lpm_disable_count = 1;
 	atomic_set(&dev->urbnum, 0);
diff --git a/drivers/usb/host/fsl-mph-dr-of.c b/drivers/usb/host/fsl-mph-dr-of.c
index da2fab89daad..c1d25163d204 100644
--- a/drivers/usb/host/fsl-mph-dr-of.c
+++ b/drivers/usb/host/fsl-mph-dr-of.c
@@ -99,6 +99,8 @@ static struct platform_device *fsl_usb2_device_register(
 	else
 		dma_set_mask(&pdev->dev, DMA_BIT_MASK(32));
 
+	pdev->dev.of_node = ofdev->dev.of_node;
+
 	retval = platform_device_add_data(pdev, pdata, sizeof(*pdata));
 	if (retval)
 		goto error;
-- 
2.17.1

